axelrod/strategies/backstabber.py

```diff
@@ -17,52 +17,81 @@ class BackStabber(Player):
     classifier = {
         'memory_depth': float('inf'),
         'stochastic': False,
-        'makes_use_of': set(['length']),
+        'makes_use_of': {'length'},
         'long_run_time': False,
         'inspects_source': False,
         'manipulates_source': False,
         'manipulates_state': False
     }
 
-    @staticmethod
-    def strategy(opponent: Player) -> Action:
-        if not opponent.history:
-            return C
-        if opponent.defections > 3:
-            return D
-        return C
+    def strategy(self, opponent: Player) -> Action:
+        return _backstabber_strategy(opponent)
 
 
-@FinalTransformer((D, D), name_prefix=None) # End with two defections
+@FinalTransformer((D, D), name_prefix=None)  # End with two defections
 class DoubleCrosser(Player):
     """
     Forgives the first 3 defections but on the fourth
-    will defect forever. If the opponent did not defect
-    in the first 6 rounds the player will cooperate until
-    the 180th round. Defects on the last 2 rounds unconditionally.
+    will defect forever. Defects on the last 2 rounds unconditionally.
+
+    If 8 <= current round <= 180,
+    if the opponent did not defect in the first 7 rounds,
+    the player will only defect after the opponent has defected twice in-a-row.
     """
 
     name = 'DoubleCrosser'
     classifier = {
         'memory_depth': float('inf'),
         'stochastic': False,
-        'makes_use_of': set(['length']),
+        'makes_use_of': {'length'},
         'long_run_time': False,
         'inspects_source': False,
         'manipulates_source': False,
         'manipulates_state': False
     }
 
     def strategy(self, opponent: Player) -> Action:
-        cutoff = 6
-
-        if not opponent.history:
-            return C
-        if len(opponent.history) < 180:
-            if len(opponent.history) > cutoff:
-                if D not in opponent.history[:cutoff + 1]:
-                    if opponent.history[-2:] != [D, D]:  # Fail safe
-                        return C
-        if opponent.defections > 3:
-            return D
+        if _opponent_triggers_alt_strategy(opponent):
+            return _alt_strategy(opponent)
+        return _backstabber_strategy(opponent)
+
+
+def _backstabber_strategy(opponent: Player) -> Action:
+    """
+    Cooperates until opponent defects a total of four times, then always defects.
+    """
+    if not opponent.history:
         return C
+    if opponent.defections > 3:
+        return D
+    return C
+
+
+def _alt_strategy(opponent: Player) -> Action:
```

Comment:
actually.  this brings up a problem i had, to which module methods was the most expedient solution.  

as i see it, these should all be private methods of their classes and DoubleCrosser should inherit from BackStabber.  The tests, as they are, should fully cover all those methods (i don't have a coverage tool, but i think i got it). 

I originally wrote `class DoubleCrosser(BackStabber):` , but strange things would happen with `@FinalTransformer((D, D), name_prefix=None) `  If only BackStabber had that decorator, DoubleCrosser would not inherit that behavior.  If both had that decorator I got a maximum recursion depth error.